xchange-blockchain/src/main/java/org/knowm/xchange/blockchain/BlockchainAdapters.java

```diff
@@ -0,0 +1,256 @@
+                .instrument(currencyPair)
+                .limitPrice(blockchainMarketDataOrder.getPrice())
+                .originalAmount(blockchainMarketDataOrder.getQuantity())
+                .build();
+    }
+
+    public static Trades toTrades(List<BlockchainTrade> blockchainTrades, CurrencyPair currencyPair) {
+        List<Trade> trades = blockchainTrades.stream()
+                .map(blockchainTrade -> {
+                    Order.OrderType orderType = blockchainTrade.isBuyer() ? Order.OrderType.BID : Order.OrderType.ASK;
+                    return new Trade.Builder()
+                            .instrument(currencyPair)
+                            .type(orderType)
+                            .originalAmount(blockchainTrade.getQuantity())
+                            .price(blockchainTrade.getPrice())
+                            .timestamp(blockchainTrade.getTimestamp())
+                            .id(blockchainTrade.getId())
+                            .build();
+                }).collect(Collectors.toList());
+        return new Trades(trades, Trades.TradeSortType.SortByTimestamp);
+    }
+
+    public static Integer currencyPairToId(CurrencyPair currencyPair) {
+        if (CurrencyPair.ETH_BTC.equals(currencyPair)) {
+            return 3;
+        }
+        throw new IllegalArgumentException(String.format("Unsupported currency pair '%s'", currencyPair));
+    }
+
+
+    public static AddressWithTag toAddressWithTag(BlockchainAccount blockchainAccount){
+        return new AddressWithTag(blockchainAccount.getAddress(), null);
+    }
+
+    public static Order.OrderStatus toOrderStatus(String status) {
+        switch (status.toUpperCase()) {
+            case "NEW":
+                return Order.OrderStatus.NEW;
+            case "FILLED":
+                return Order.OrderStatus.FILLED;
+            case "EXPIRED":
+                return Order.OrderStatus.EXPIRED;
+            case "CANCELED":
+                return Order.OrderStatus.CANCELED;
+            case "REJECTED":
+                return Order.OrderStatus.REJECTED;
+            case "PENDING_CANCEL":
+                return Order.OrderStatus.PENDING_CANCEL;
+            case "PARTIALLY_FILLED":
+                return Order.OrderStatus.PARTIALLY_FILLED;
+            default:
+                return Order.OrderStatus.UNKNOWN;
+        }
+    }
+
+    public static OpenOrders toOpenOrders(List<BlockchainOrder> blockchainOrders){
+        List<LimitOrder> limitOrders = new ArrayList<>(Collections.emptyList());
+        List<Order> hiddenOrders = new ArrayList<>(Collections.emptyList());
+
+        for(BlockchainOrder blockchainOrder : blockchainOrders) {
+            final Order.OrderType orderType = blockchainOrder.isBuyer() ? Order.OrderType.BID : Order.OrderType.ASK;
+            final CurrencyPair symbol = blockchainOrder.getSymbol();
+            Order.Builder builder;
+
+            if (blockchainOrder.isMarketOrder()) {
```

Comment:
extract this code to another method and call it something like "createOrderBuilder" for better legibility